packages/babel-plugin-react-jsx/src/TransformJSXToReactBabelPlugin.js

```diff
@@ -24,6 +24,42 @@
 'use strict';
 
 const esutils = require('esutils');
+const helperModuleImports = require('@babel/helper-module-imports');
+const isModule = helperModuleImports.isModule;
+const addNamespace = helperModuleImports.addNamespace;
+const addNamed = helperModuleImports.addNamed;
+const addDefault = helperModuleImports.addDefault;
+
+// These are all the valid auto import types (under the config autoImport)
+// that a user can specific
+const IMPORT_TYPES = {
+  none: 'none', // default option. Will not import anything
+  require: 'require', // var _react = require("react");
+  namespace: 'namespace', // import * as _react from "react";
+  default: 'default', // import _default from "react";
+  namedExports: 'namedExports', // import { jsx } from "react";
+};
+
+// We want to use React.createElement, even in the case of
+// jsx, for <div {...props} key={key} /> to distinguish it
+// from <div key={key} {...props} />. This is an intermediary
+// step while we deprecate key spread from props. Afterwards,
+// we will remove createElement entirely
+function shouldUseCreateElement(path, t) {
```

Comment:
```suggestion
function shouldUseCreateElement(path, types) {
```
I wasn't sure what `t` meant here without looking at it's callsite.